invenio_app_ils/records/api.py

```diff
@@ -65,7 +91,176 @@ def update(self, data):
+        raise RelatedRecordError(
+            "No relation type with id: {}".format(relation_id)
+        )
+
+    @property
+    def editions(self):
+        """Get related editions."""
+        return self._get_related(self.edition_relation())
+
+    @property
+    def languages(self):
+        """Get related languages."""
+        return self._get_related(self.language_relation())
+
+    def node(self, relation_type):
+        """Get PIDNode for this record."""
+        return PIDNode(pid=self.pid, relation_type=relation_type)
+
+    def add_related_edition(self, record):
+        """Add record as a related edition record."""
+        self.add_related(record, self.edition_relation())
+
+    def remove_related_edition(self, record):
+        """Remove related edition record."""
+        self.remove_related(record, self.edition_relation())
+
+    def add_related_language(self, record):
+        """Add record as a related language record."""
+        self.add_related(record, self.language_relation())
+
+    def remove_related_language(self, record):
+        """Remove related language record."""
+        self.remove_related(record, self.language_relation())
+
+    def add_related(self, record, relation_type):
+        """Add related record with a relation."""
+        self._add_related_record(
+            record,
+            relation_type,
+            parent_node=self._parent_node(relation_type)
+        )
+
+    def remove_related(self, record, relation_type):
+        """Remove related record with a relation."""
+        self._remove_related_record(
+            record,
+            relation_type,
+            parent_node=self._parent_node(relation_type)
+        )
+
+    def _add_related_record(self, record, relation_type, parent_node=None):
+        """Add related record with a relation."""
+        record_node = record.node(relation_type)
+        if record_node.is_parent or record_node.is_child:
+            raise RelatedRecordError(
+                "failed to add related record because it already has relations"
+            )
+        if parent_node is None:
+            parent_node = self.node(relation_type)
+        parent_node.insert_child(record.pid)
+        self.dump_related_add(record, relation_type, update_related=True)
+
+    def _remove_related_record(self, record, relation_type, parent_node=None):
+        """Remove related record with a relation."""
+        if parent_node is None:
+            parent_node = self.node(relation_type)
+        self.dump_related_remove(record, relation_type, update_related=True)
+        parent_node.remove_child(record.pid)
+
+    def dump_related_add(self, record, relation, update_related=False):
+        """Add record to related records dump."""
+        obj = self.record_to_dump(record, relation)
+        related_records = self["related_records"]
+        if obj not in related_records:
+            related_records.append(obj)
+
+            if update_related:
```

Comment:
not 100% sure about this extra If.
Shouldn't it happen all the time? Why `false` by default?

Reply:
When adding/removing relations it'll dump the current record then go to all its related records and dump those too with the new relation. If we don't set it to stop update it's related records it'll end up in an infinite loop.

There was no particular reason for it to be `false` by default, probably makes more sense to keep is as a positional argument.